target/arc/translate.c

```diff
@@ -819,6 +819,11 @@ arc_gen_SYNC(DisasCtxt *ctx)
     return ret;
 }
 
+int
+arc_gen_DSYNC(DisasCtxt *ctx)
+{
+    return DISAS_UPDATE;
```

Comment:
I think we should use `tcg_gen_mb` here for data sync. I would suggest using at least  ` TCG_MO_ALL`.

Reply:
You are very right. I wasnt aware of that capability in QEMU. I also took the liberty of adding that memory barrier, albeit more strict, to the SYNC instruction